packages/material-ui/src/Select/SelectInput.js

```diff
@@ -321,7 +321,7 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
         aria-labelledby={`${labelId || ''} ${buttonId || ''}`}
         aria-haspopup="listbox"
         onKeyDown={handleKeyDown}
-        onClick={disabled || readOnly ? null : handleClick}
+        onMouseDown={disabled || readOnly ? null : handleClick}
```

Comment:
Could you rename this to `handleMouseDown` and make sure click isn't called from that one?

The tests wouldn't catch this double invokation since they don't simulate all the events that are dispatch when the user clicks with a mouse.